src/amo/components/Collection/index.js

```diff
@@ -94,17 +94,36 @@ export type SaveAddonNoteFunc = (
 ) => void;
 
 export class CollectionBase extends React.Component<InternalProps> {
+  addonPlaceholderCount: number;
+
   static defaultProps = {
     _config: config,
     _isFeaturedCollection: isFeaturedCollection,
     creating: false,
     editing: false,
   };
 
+  constructor(props: InternalProps) {
+    super(props);
+    this.addonPlaceholderCount = 5;
+    this.maybeResetAddonPlaceholderCount();
+  }
+
+  maybeResetAddonPlaceholderCount() {
+    const { collection } = this.props;
+    if (collection && collection.addons && collection.addons.length) {
```

Comment:
It seems like this could just be: 

`this.addonPlaceholderCount = collection && collection.addons && collection.addons.length || 5;`

and you could then do away with setting the default in `constructor`. Maybe using a `const` in place of the `5` would be a good idea in this case.

Reply:
This won't work because when `loading=true`, `collection` is null which means it will reset the value back to 5. We want it to use the previous value instead.

This whole approach is a little bit un-react-y in that it relies on the state of the component. The `render()` cycle is not working with the current value of `addonPlaceholderCount`, it's working with the *previous* value. This is just one of those patterns that's hard to achieve cleanly in React.

Reply:
I added a comment to this code to make it more clear.